### PR TITLE
fix: correct tie-breaking logic in breakTies function

### LINKED ISSUE #329: bug: Tie-breaking ignores numerator filter due to discarded .filter() result
In `src/computation/logic/utils.ts`, `breakTies()` tries to narrow the pool of tied parties to those with the highest original vote count (numerator) before doing a random draw:

```ts
winnersCopy.filter((item) => baseValue[item.key] === maxNumerator);
```

The result of `.filter()` is never assigned back, so `winnersCopy` still contains all tied parties. The coin flip on the next line therefore picks randomly among all tied parties, ignoring the numerator entirely.

The fix is straightforward:

```ts
const narrowed = winnersCopy.filter((item) => baseValue[item.key] === maxNumerator);
return narrowed[Math.floor(Math.random() * narrowed.length)];
```

### PR DESCRIPTION
**Pull request by Log234**
This pull request fixes a bug in the `breakTies` function in `src/computation/logic/utils.ts` to ensure that the random selection is only performed among the actual top-scoring winners, not the full original list.

Bug fix:
* Corrected the random selection logic in `breakTies` so that it only selects from the filtered list of winners with the highest numerator, rather than from the unfiltered list.

## How to test
We do not actually have any election results that would be impacted by this change, as it's such a rare edge case that none of our voting data meets the requirements regardless of the settings chosen.

## Issues closed
Closes #329

### CODE REVIEW

File: src/computation/logic/utils.ts [hunk 1]
```diff
@@ -318,9 +318,9 @@ export function breakTies(winners: KeyValuePair[], baseValue: _.Dictionary<numbe
     const maxNumerator = Math.max(...numerators);
 
     // Filter out all winners that did not have the highest numerator
-    winnersCopy.filter((item) => baseValue[item.key] === maxNumerator);
+    const narrowed = winnersCopy.filter((item) => baseValue[item.key] === maxNumerator);
```
Comment (simendag) [inline src/computation/logic/utils.ts:321]: Haha, I guess we intended to do this, but didn't actually use the result?

**Review by simendag — APPROVED** [review]
🚀
